### ISSUE TITLE
How does the defined feature contribute to regression?

### ISSUE THREAD

**Issue by henryJhonOt** [yewzijian/RPMNet#31]
I'm here...again....
My work generally followed your backbone with a few modification which includes replacing `ppf` feature and loss function.
I defined a `pmd` feature and it worked pretty well on clean data, but failed in `jitter` and `crop` in the first 200 epoch.
I guess my `pmd` feature is not robust enough to partial or noisy data. But I'm totally confused how to improve the robustness of a local feature.

Here's the brief definition of `pmd`. Given point `S`, select `k` neighbors `K_i`, mark `A = K_1, B=K_2, C=K_3`. Thus `S, A, B, C` forms a triangular cone `S-ABC` where `S` is the appex. Three meshes, `SAB, SAC, SBC`, intersect at point `S`. 
Thus, 
```python
# mesh SAB, SAC, SBC intersect at S

angle_SA=angle(SAB, SAC)
angle_SB=angle(SAB, SBC)
angle_SC=angle(SAC, SBC)

# ni, nr, d_norm are the same as RPMNet
angle_ni = angle(ni, nr)
d_norm = torch.norm(d, dim=-1)

# SOME_CODES_HERE
# angle_SA, angle_SB, angle_SC  are repeated here to keep a same shape with nr_ni and d_norm

pmd = torch.stack([angle_SA, angle_SB, angle_SC, nr_ni, d_norm], dim=-1)
return {'xyz': new_xyz, 'dxyz': xyz_feat, 'pmd': pmd}
```

_**I'm not asking for detailed implementation**_...That's quite rude...
It would be nice of you if you could tell me:
Did you try other features before `ppf`, especially any features designed by yourself?
Is it possible to improve the robustness of handcraft features during its definition, not during training afterwards.

**Comment by henryJhonOt** [comment]
By the way, I tried set `num_neighbors=3`, slower as it is, `ppf` still achieves the same accuracy with more epochs.
But in this case, `pmd` is much worse. 
So definitely something's wrong with `pmd`.

**Comment by yewzijian** [comment]
Hello, I didn't try any additional features. The PPF is a natural choice since it comes from PPFNet, which was based on traditional descriptors (PFH/FPFH).

Your pmd features appears to make sense. However, unlike the point pair features (which considers two points), you consider 4 points (center + 3 other points) at one time. This may be problematic, e.g., how do you order the points A,B,C consistently? Especially in the noisy scenario.

**Comment by henryJhonOt** [comment]
> Hello, I didn't try any additional features. The PPF is a natural choice since it comes from PPFNet, which was based on traditional descriptors (PFH/FPFH).
> 
> Your pmd features appears to make sense. However, unlike the point pair features (which considers two points), you consider 4 points (center + 3 other points) at one time. This may be problematic, e.g., how do you order the points A,B,C consistently? Especially in the noisy scenario.

Hello. I really really appreciate your help since I wathed this repo....Here's the progress since I opened this issue.
I figured out a way to transform the Partial-Partial registration to Clean registration via selecting the shared points between src and ref and it worked very well. But I'm not sure whether it is a little tricky since it is not a Patial-Partial registration. Nevertheless, leave this concern to me.
On Noise registration, less accurate than RPMNet as it is, I managed to outperform other SOTA.

Thank you again for your help for a long time. And I'm re-thinking about pmd on your proposed problem.
